app/uk/gov/hmrc/tradergoodsprofilesrouter/controllers/action/ValidationRules.scala

```diff
@@ -114,13 +115,24 @@ trait ValidationRules {
 
   private def validateWithdrawReasonQueryParam(implicit
```

Comment:
we can get rid of the nested statement and case, you will need to change the validateWithdrawAdviceQueryParam function a little and controller as the suggestion below return a Either[Error, Option[String]] 

```suggestion
  private def validateWithdrawReasonQueryParam(implicit
                                               request: Request[JsValue]
                                              ): Either[Error, Option[String]] = {
    Try(request.body.as[WithdrawReasonRequest])
      .toOption match {
      case Some(v)  if v.withdrawReason.exists(_.isEmpty) =>
        Left(Error(InvalidQueryParameter, InvalidWithdrawReasonNullMessage, invalidWithdrawReasonCode))
      case Some(v) if v.withdrawReason.exists(_.length > WithdrawReasonCharLimit) =>
        Left(Error(InvalidQueryParameter, invalidWithdrawReasonMessage, invalidWithdrawReasonCode))
      case Some(v) => Right(v.withdrawReason)
      case _ =>  Right(None)

    }
  }